### ISSUE TITLE
Inconsistency SSE Type and REST State

### ISSUE THREAD

**Issue by sja** [eclipse/smarthome#3282]
**tl;dr**
As a generic UI I have no chance to find out, which type the current state of an DimmerItem has, until I receive an event-message via SSE with that information. Thats a problem at initial rendering because I cannot trigger that event-message.

**Details**
I hope this is an understanding problem at my side. I tried to find an issue related to this but it is not easy to search for that topic.

I have an Dimmer item defined, which state is `24`. Now I do a HTTP POST with a body `26`, then it has the new state `26`. Until here as expected.

Now I POST a `0.27` and my UI interprets it as 0.27%, but another UI interprets it as 27%.

If I retrieve the item's state via the REST API, I have no way of knowing the type of that state value - because that type is only provided via the event stream (SSE).

All I really need is an additional field in the JSON representation that corresponds to the event payload's type information.

Another thing here is that I cannot provide a type information with the REST call, but the SSE Events contain a Type (`PercentType`) after the state changes/updates. That type also never changes.

A solution would be to provide that type information on the Item-endpoint JSON (ItemResource) as sibling to the `state` property.
So that additional field in the JSON representation should correspond to the event payload's type information.

Examples:
```
$ curl -X POST -H 'Content-Type: text/plain' --data-binary 70 http://localhost:8080/rest/items/DemoRoom_Dimmer
→ { type: "PercentType", value: "70" }
```

```
$ curl -X POST -H 'Content-Type: text/plain' --data-binary 0.6 http://localhost:8080/rest/items/DemoRoom_Dimmer
→ { type: "PercentType", value: "0.6" }
```

```
curl -X POST -H 'Content-Type: text/plain' --data-binary ON http://localhost:8080/rest/items/DemoRoom_Dimmer
→ { type: "PercentType", value: "100" }
```

**Comment by triller-telekom** [comment]
I just had a look at the SSE Eventstream but I cannot find the place where it inserts the `type` of the state. To me it looks the same as if I query the `ItemResource` via rest call.

In both cases an `EnrichedItemDTO` object is created and this does not have any type information on the state. So why do you think the SSE stream has this information?

**Comment by FND** [comment]
I can't comment on the Java implementation behind the scenes, but here's what being sent over the wire:

```
$ curl http://demo.openhab.org:8080/rest/events
event: message
data: {"topic":"smarthome/items/Sunrise_Time/state","payload":"{\"type\":\"DateTimeType\",\"value\":\"2017-05-16T03:10:00.000+0000\"}","type":"ItemStateEvent"}

…

event: message
data: {"topic":"smarthome/items/Sun_Azimuth/state","payload":"{\"type\":\"DecimalType\",\"value\":\"243.13\"}","type":"ItemStateEvent"}

…

event: message
data: {"topic":"smarthome/items/Moon_Phase/state","payload":"{\"type\":\"StringType\",\"value\":\"WANING_GIBBOUS\"}","type":"ItemStateEvent"}
```

As you can see, there we do have type information (in this case: `DateTimeType`, `DecimalType` and `StringType`).

That same information was not included in the item representations last time we'd checked (not sure whether we were using an outdated version; I'm afraid I can't check right now). However, that appears to have changed since:

```
$ curl http://demo.openhab.org:8080/rest/items/Sunrise_Time | grep type
  "type": "DateTime",
$ curl http://demo.openhab.org:8080/rest/items/Sun_Azimuth | grep type
  "type": "Number",
$ curl http://demo.openhab.org:8080/rest/items/Moon_Phase | grep type
  "type": "String",
```

(Note, though, that the identifiers are slightly inconsistent with the event stream's, omitting the "Type" suffix.)

So @sja, inconsistencies aside, I guess we now have everything we need?

**Comment by sja** [comment]
A bit inconsistent because of that "Type" suffix. But yes, then we have everything together. 
We used 2.1.0-SNAPSHOT build 839.

**Comment by maggu2810** [comment]
> A bit inconsistent because of that "Type" suffix

@kaikreuzer @sjka Any change to make this consistent? We will break the API and so every UI and other consumers of the event stream.
We removed the `Type` postfix sometime on the REST API, too.
So I think it is a way to go in front of a 1.0 version.

**Comment by kaikreuzer** [comment]
@maggu2810 I agree, this is inconsistent and was probably missed when we changed the ItemResource. It looks as if `ItemEventFactory` is the one to address as well as this uses the simple class name of the type, which leaves the "Type" postfix in place.
